server/server.js

```diff
@@ -1,13 +1,76 @@
-// make this a simple http server that writes to the response stream object
+/*
+ * Homework 
+ * JSCRIPT300-Spring2015/Module_3
+ * by Diane Zevenbergen
+ */
 
-// the page should display the following message:
+'use strict';
 
-// Today is <day name>, <month name> <date>. Here are the available food trucks:
+// Question: Is there a way I can avoid using 'output' as a global var?
+var output = [];
```

Comment:
Yes, just declare the variables you need in the request handler. Your request handler can call the filterByDay method on the trucks module. Rather than making the createServer call inside another function declaration, just use the event callback structures to help you out. You can either A. `http.createServer(function (request, response) { });` and inside your callback call another function that returns the string, or, if you want to separate them out, do `var server = createServer(); server.on('request', function (request, response) { });` The function callbacks give you the scopes you need to keep the array inside a function scope.